### PR TITLE
DD-1356: map depositorId to dataSupplier

### PR DESCRIPTION
**Pull request by jo-pol**
Fixes DD-1356: map depositorId to dataSupplier

# Description of changes

# How to test

* build,  also dd-validate-dans-bag after `rm -rf application api`
* `start-preprovisioned-box.py dev_transfer`
* `deploy.py -m dd-vault-ingest-flow dev_vaas` as well as `dd-validate-dans-bag`
* further instructions under set-up in `dev.md`

Checks on dev_transfer:
  * [x] DB shows user name (`sudo su postgres`; `psql dd_vault_catalog`; `select bag_id, data_supplier from ocfl_object_versions`)
  * [x] `/var/opt/dans.knaw.nl/tmp/ocfl-tar/inbox/vaas-<UUID>-v2-ttv2.zip` shows the user name in `metadata/oai-ore.*`

The fully local test shows the same DB values and ZIP details.

# Related PRs
(Add links)
*

# Notify
@DANS-KNAW/dataversedans

### CODE REVIEW

File: docs/dev.md [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+===========
+This page contains information for developers about how to contribute to this project.
+
+Set-up
+------
+This project can be used in combination with  [dans-dev-tools]{:target=_blank}. 
+Before you can start it as a service some dependencies must first be started.
+For one dependency you have the choice between a Virtual Machine (VM) `dev_transfer`
+and a local service `dd-vault-catalog`. For the VM you need access to the project [dd-dtap]{:target=_blank},
+the local service requires a local database.
```
Comment (jo-pol) [inline docs/dev.md]: So far, trying with the local vault-catalog resulted in an exception. @janvanmansum Drop this option for now?

```
ERROR [2023-09-05 14:20:43,390] [vault-ingest-worker-1] nl.knaw.dans.vaultingest.client.VaultCatalogClient: Error while registering deposit: {"code":404,"message":"HTTP 404 Not Found"}
! nl.knaw.dans.vaultcatalog.client.ApiException: {"code":404,"message":"HTTP 404 Not Found"}
! at nl.knaw.dans.vaultcatalog.client.ApiClient.invokeAPI(ApiClient.java:1067)
! at nl.knaw.dans.vaultcatalog.client.OcflObjectVersionApi.createOcflObjectVersionWithHttpInfo(OcflObjectVersionApi.java:108)
! at nl.knaw.dans.vaultcatalog.client.OcflObjectVersionApi.createOcflObjectVersion(OcflObjectVersionApi.java:68)
! at nl.knaw.dans.vaultingest.client.VaultCatalogClient.registerDeposit(VaultCatalogClient.java:58)
```

**Review by janvanmansum — CHANGES_REQUESTED** [review]
See my amendment PR on your topic branch.

File: docs/dev.md [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+
+```commandline
+start-env.sh
+```
+
+The service `dd-validate-dans-bag` needs different configurations for `dd-vault-ingest-flow` and other services. 
+So you will have to update the generated `dd-validate-dans-bag/etc/config.yml`,
+perhaps keep copies of the files or comment lines to switch between both situations.
+
+* remove one level from `../../` in the schema locations.
```
Comment (PaulBoon) [inline docs/dev.md]: I only see ../dans-schema, so it might not be needed

File: docs/dev.md [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+        depositor.userId = user001
+        bag-store.bag-name = default-open
+
+    Note that 
+    * The `bag-name` should match the copied bag.
+    * The `userId` should match a value configured as a `dataSupplier` in `dd-vault-ingest-flow/etc/config.yml`.
+    * The `<UUID>` should match the directory name
+* When using the deposit on a VM, create it at `dd-dtap/shared/<UUID>`, on the VM
+
+      cd /vagrant/shared/<UUID> ; chgrp -R deposits . ; chown -R dd-vault-ingest-flow .
```
Comment (PaulBoon) [inline docs/dev.md]: chown en chgrp did not work; ownership remained vagrant:vagrant.

File: src/main/java/nl/knaw/dans/vaultingest/core/deposit/Deposit.java [hunk 4]
```diff
@@ -123,6 +124,7 @@ public String getBagId() {
     }
 
     public Collection<DepositFile> getPayloadFiles() {
+        // TODO why these inconsistent names, it seems to prevent @Getter
```
Comment (PaulBoon) [inline src/main/java/nl/knaw/dans/vaultingest/core/deposit/Deposit.java:127]: The 'TODO' action might be to consider renaming, make it explicit and say so.

File: src/main/java/nl/knaw/dans/vaultingest/core/inbox/AutoIngestArea.java [hunk 5]
```diff
@@ -28,16 +29,18 @@ public class AutoIngestArea {
...
     private final DepositToBagProcess depositToBagProcess;
     private final Outbox outbox;
+    private final Map<String, String> dataSupplierMap;
 
     public AutoIngestArea(
         Executor executor,
         IngestAreaWatcher ingestAreaWatcher,
         DepositToBagProcess depositToBagProcess,
-        Outbox outbox) {
+        Outbox outbox, Map<String, String> dataSupplierMap) {
```
Comment (PaulBoon) [inline src/main/java/nl/knaw/dans/vaultingest/core/inbox/AutoIngestArea.java]: Please start the Map argument on a new line because is more consistent.

File: src/main/java/nl/knaw/dans/vaultingest/core/inbox/MigrationIngestArea.java [hunk 6]
```diff
@@ -33,17 +34,18 @@ public class MigrationIngestArea {
...
     private final Outbox outbox;
+    private final Map<String, String> dataSupplierMap;
 
     public MigrationIngestArea(
         ExecutorService executorService,
         DepositToBagProcess depositToBagProcess,
         Path inboxPath,
-        Outbox outbox
-    ) {
+        Outbox outbox, Map<String, String> dataSupplierMap) {
```
Comment (PaulBoon) [inline src/main/java/nl/knaw/dans/vaultingest/core/inbox/MigrationIngestArea.java]: Same here, just a newline before the Map.

File: src/test/java/nl/knaw/dans/vaultingest/core/ConfigurationTest.java [hunk 7]
```diff
@@ -0,0 +1,70 @@
...
+        File cfgFile = testDir.resolve("config.yml").toFile();
+        FileUtils.write(cfgFile, rewriteFilePaths("src/test/resources/debug-etc/config.yml"), StandardCharsets.UTF_8);
+        assertDoesNotThrow(() -> factory.build(FileInputStream::new, cfgFile.toString()));
+    }
+
+    private String rewriteFilePaths(String pathname) throws IOException {
+        // prevent ConfigurationValidationException ... does not exist but is required
+        // TODO why not @FileMustExist for other files and dirs?
+        return FileUtils.readFileToString(new File(pathname), StandardCharsets.UTF_8)
+            .replaceAll("(iso639[12]): .*/","$1: src/main/assembly/dist/cfg/");
```
Comment (PaulBoon) [inline src/test/java/nl/knaw/dans/vaultingest/core/ConfigurationTest.java:68]: iso639 are language codes, what does this have to do with file paths?
So either change the name, because it is now very general, or add more comment that explains that is has to do with the language section of the config...

File: src/test/java/nl/knaw/dans/vaultingest/core/deposit/CommonDepositManagerIntegrationTest.java [hunk 8]
```diff
@@ -33,10 +35,21 @@ void loadDeposit() {
...
+        var deposit = manager.loadDeposit(Path.of(s.getPath()), Map.of("user001","Name of user"));
         assertThat(deposit.getId()).isEqualTo("0b9bb5ee-3187-4387-bb39-2c09536c79f7");
     }
 
+    @Test
+    void loadDeposit_should_complain_about_missing_data_supplier() {
+        var manager = new DepositManager(new XmlReader());
+
+        var s = getClass().getResource("/input/0b9bb5ee-3187-4387-bb39-2c09536c79f7");
+        assert s != null;
```
Comment (PaulBoon) [inline src/test/java/nl/knaw/dans/vaultingest/core/deposit/CommonDepositManagerIntegrationTest.java]: Why are we not using assertNotNull(s) ?

File: docs/dev.md [hunk 9]
```diff
@@ -0,0 +1,119 @@
...
+and the database: 
+
+    sudo su postgres
+    psql dd_vault_catalog
+    select bag_id, data_supplier from ocfl_object_versions;
+    \c dd_transfer_to_vault
+    select bag_id, data_supplier from transfer_item;
+
+To examine the local database run `start-hsqldb-client.sh`
+and connect with the `database.url` specified in `dd-vault-catalog/etc/config.yml`,
```
Comment (PaulBoon) [inline docs/dev.md]: Better provide a tip on how to inspect the database, took me some time to discover that I could run the 'manager' with `java -cp ~/git/service/data-station/dd-vault-catalog/data/hsqldb-2.7.1.jar org.hsqldb.util.DatabaseManagerSwing`.

File: docs/dev.md (hunk 2 above)
Comment (jo-pol) [reply docs/dev.md]: That change was pretty recent.

File: docs/dev.md (hunk 3 above)
Comment (jo-pol) [reply docs/dev.md]: changed into a simpler `chmod -R 777`
